data/processing.py

```diff
@@ -163,27 +164,30 @@ def run(date):
   # Overwrites `domains` and `subdomains` in-place.
   process_domains(domains, agencies, subdomains, parent_scan_data, subdomain_scan_data)
 
+  _app = create_app('production')
+
   # Reset the database.
   LOGGER.info("Clearing the database.")
-  models.clear_database()
-
-  # Calculate agency-level summaries. Updates `agencies` in-place.
-  update_agency_totals(agencies, domains, subdomains)
-
-  # Calculate government-wide summaries.
-  report = full_report(domains, subdomains)
-  report['report_date'] = date
-
-  LOGGER.info("Creating all domains.")
-  Domain.create_all(domains[domain_name] for domain_name in sorted_domains)
-  LOGGER.info("Creating all subdomains.")
-  Domain.create_all(subdomains[subdomain_name] for subdomain_name in sorted_subdomains)
-  LOGGER.info("Creating all agencies.")
-  Agency.create_all(agencies[agency_name] for agency_name in sorted_agencies)
-
-  # Create top-level summaries.
-  LOGGER.info("Creating government-wide totals.")
-  Report.create(report)
+  with _app.app_context() as ctx:
```

Comment:
This is needed now that the `models.py` code is tied in more with the fact that this is a flask app, so things happen within the context of a request. The change to add that coupling was done to be able to more easily have a database connection that was integrated into the flask app in such a way that a connection didn't need to be created for each incoming request.